### ISSUE TITLE
Don't quit if there was a problem obtaining a cert for one domain

### ISSUE THREAD

**Issue by ponychicken** [mholt/caddy#971]
(Are you asking for help with using Caddy? Please use our forum instead: https://forum.caddyserver.com. If you are filing a bug report, please answer the following questions. If your issue is not a bug report, you do not need to use this template. Either way, please consider donating if we've helped you. Thanks!)

#### 1. What version of Caddy are you running (`caddy -version`)?
Caddy 0.9.0

#### 2. What are you trying to do?
I'm trying to adapt Caddy for usage with mail-in-a-box. ("Mail-in-a-Box helps individuals take back control of their email by defining a one-click, easy-to-deploy SMTP+everything else server: a mail server in a box. ")

Often users will not yet have setup the correct DNS records before starting the install of mail-in-a-box. In this case Caddy will fail to obtain a certificate (as expected). Unfortunately Caddy doesn't simply serves the sites for which it either has a certificate or doesn't need one (ip adresses), but quits after it failed to obtain the certificate. This makes it impossible to access the site even via the ip address.

#### 3. What is your entire Caddyfile?
(IP and domain obcured)
```text
XXX.XXX.XXX.XXX:80 {
        # Expose this directory as static files.
        root /home/user-data/www/default
}

mail.example.com {
        tls {
                max_certs 50
        }

        # Expose this directory as static files.
        root /home/user-data/www/default
}


```

#### 4. How did you run Caddy (give the full command and describe the execution environment)?
Using the upstart script.

#### 5. What did you expect to see?
For Caddy to start and serve the content of the root folder specified for the IP address.
I assumed that the on-demand certs might help, but the problem is the same.

#### 6. What did you see instead (give full error messages and/or log)?
Caddy quits and is then continually restarted by the init script.

**Comment by ponychicken** [comment]
A simpler repo case would be this:

````
localhost, somevaliddomain.com, anothervaliddomain.org, invalid.com {
        browse
}
````

run with ``caddy --conf=Caddyfile``

**Comment by mholt** [comment]
Thanks for simplifying the repro! Love to see that. :+1:

A few things are going on here, but they are not bugs in Caddy. Still, thanks for raising the issue so I can reference this in future discussions.

I feel pretty strongly about Caddy reporting errors, especially what are essentially critical errors like "My core competency -- the whole reason you downloaded me -- is serving websites. I can't serve this site because its security is broken, and I can't fix it for you." Merely logging is not sufficient for these errors while the operator is still present, so the process quits, and rightly so. It'll take a lot to sway my opinion on this matter. Caddy serves all or nothing.

Caddy being continually being restarted by the init script is a bug in the init script (which I don't support, sorry -- but its maintainer might) -- I wouldn't recommend trying to restart with the same broken configuration over and over. That doesn't make much sense. :confused:

Basically, only tell Caddy to serve sites that are ready to serve. If a site isn't ready to serve, don't try to serve it.
